Labs 15 staff can create marketing information for toggle

vokoblin opened the pull request:
All the marketing information is optional and if not provided will me set to empty string.
Gets send to the gateway where will be extracted and send to DB.

Commit:
adds first test

Commit:
adds marketing variables

Commit:
adds classeroo

Commit:
Completes CreateMarketableInformationForToggle.

Commit:
Travis please forgive me.

Commit:
Merge branch 'master' of github.com:clearbooks/labs into LABS-15_Staff_can_CreateMarketingInformationForToggle

Commit:
Adds CreateMarketingInformationRequest class
Functions as a container for marketing information.

Commit:
re-Scruyinizer

RWood1 commented:
gosh this has been open a while, I'll give this a quick review for you

RWood1 commented on src/Toggle/CreateMarketingInformationForToggle.php:
```diff
@@ -0,0 +1,39 @@
+    public function execute( CreateMarketingInformationRequest $request )
+    {
+
+        $this->gateway->setMarketingInformationForToggle( $request );
```
I don't think I would pass the request straight to the gateway, The gateway shouldn't need to know about how the use case is called. I think two parameters, 1) the toggle id and 2) an array in the format of [ 'column_name' => 'value' ]

RWood1 commented on test/Toggle/CreateMarketingInformationForToggleTest.php:
```diff
@@ -0,0 +1,101 @@
+     */
+    private function createMarketingInformationFroToggle( $gateway, CreateMarketingInformationRequest $request )
+    {
+        ( new CreateMarketingInformationForToggle( $gateway ) )->execute( $request );
```
You pass in the same gateway every time so you don't really need to pass it at all, you can just create it each time.

In fact I would create the CreateMarketingInformationForToggle object and the spy gateway object in the setUp method to simplify things

Commit:
Makes CreateMarketingInformationRequest into
Concrete entity and creates MarketingInformationRequest Interface
making all the tests pass with this structure.

Commit:
Changing the execute method to accept array of marketing information and a
toggleId. Aslo making a tests to work with this structure.
moving MarketableTogglegatewaySpy to setUp.

Commit:
Removes un-used vars

RWood1 commented on src/Toggle/CreateMarketingInformationForToggle.php:
```diff
@@ -0,0 +1,50 @@
+            'screenshot_urls' => $request->getImageLink(),
+            'discription_of_toggle' => $request->getDescriptionOfToggle(),
+            'discription_of_functionality' => $request->getDescriptionOfFunctionality(),
+            'discription_of_benefits_and_reasonForImplementation' => $request->getDescriptionOfReasonForImplementation(),
```
You swap between underscores and camel case here, column names usually have underscores so stick with that. Its quite long though how about just `description_of_implementation_reason`

Also you have spelt 'Description' wrong :wink:

Commit:
spell check and renaming

RWood1 commented on test/Toggle/CreateMarketingInformationForToggleTest.php:
```diff
@@ -0,0 +1,112 @@
+     */
+    private function createMarketingInformationFroToggle( $gateway, MarketingInformationRequest $request )
+    {
+        ( new CreateMarketingInformationForToggle( $gateway ) )->execute( $request );
```
instead of `$gatway` being a parameter, you could just call `$this->gateway`

Commit:
calling ->gateway
